sdk/playwrighttesting/microsoft-playwright-testing/src/common/httpService.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import axios, { AxiosRequestConfig, AxiosResponse, Method } from "axios";
+import { randomUUID } from "crypto";
+import { Agent } from "https";
+
+const httpsAgent = new Agent({
+  keepAlive: true,
+  timeout: 60000,
+});
+
+export class HttpService {
+  public async callAPI(
+    method: Method,
+    url: string,
+    data: any | null,
+    token: string,
+    correlationId: string,
+  ): Promise<AxiosResponse> {
+    const config: AxiosRequestConfig = {
```

Comment:
`@azure/core-rest-pipeline` should provide all the functionality you need here. It'll also ensure we log, add distributed tracing, automatically retry, etc.

Is there a reason not to use it here? Feel free to reach out in the JS team channel or me directly if you're not sure

Reply:
moved to `@azure/core-rest-pipeline`